### ISSUE TITLE
How can i fix ther overlay with statusbar or navbar?

### ISSUE THREAD

**Issue by DarkCode01** [ionic-team/capacitor#1947]
Description of the problem:
The navbar is overlay of header os cellphone, i've trying all article on internet pages

The navbar merges with the header of the ios device, I am using reactjs in the project and when compiling it to the native, the navbar has an erroneous position.

Affected platform
- [ ] Android
- [* ] iOS
- [ ] electron
- [ ] web

OS of the development machine
- [ ] Windows
- [ *] macOS
- [ ] linux

Other information:
The code of project waas make on reactjs without ionic framework
Capacitor version:
1.2.0
node version:
10.16.0
npm version:
6.9.0
CocoaPods version:
1.7.5

Steps to reproduce:
Link to sample project:

**Comment by brian-g** [comment]
Do you have `viewport-fit=contain` in your <meta name="viewport"> tag? 

With this flag you will still need to offset using Apple's env() CSS functions. https://developer.mozilla.org/en-US/docs/Web/CSS/env

**Comment by DarkCode01** [comment]
yes, i use that tag `<meta name="viewport" content="viewport-fit=contain, width=device-width, initial-scale=1">` on my index.html.

**Comment by cambronjay** [comment]
@brian-g @DarkCode01 This is an issue in the latest version of capacitor for ios. The older version does not have this issue. The app header is under the status bar and the styling is broken. You can test this by pulling down the ionic stencil conference app and running it on a device or simulator.

**Comment by jcesarmobile** [comment]
@cambronjay in which version was it broken? or in which version does it work as you expect?

can any of you provide a sample app where it can be easily reproduced? also screenshots can help

**Comment by knona** [comment]
For me it is like that.

https://github.com/ionic-team/capacitor/issues/2036#issuecomment-542287878

status bar is over the webcontent.

**Comment by jcesarmobile** [comment]
so you want apps to look like this (web content is shown below the status bar)

![IMG_2443](https://user-images.githubusercontent.com/1637892/72373293-45e31f00-3708-11ea-871b-ce5449b4b14e.PNG)
instead of like this (web content is integrated into the status bar)
![IMG_2442](https://user-images.githubusercontent.com/1637892/72373299-48457900-3708-11ea-862b-d5515f7aec32.PNG)

is that right?

capacitor has worked like this since beta 7, on the linked issue cambronjay says the bahaviour changed from 1.1.1 to 1.2.0, so still not sure if this is the issue you mean, but if that's what you are looking for, then we can probably make it configurable to also allow this behaviour.

**Comment by knona** [comment]
It is precisely that, it would be really great if you can add it !

I come from the cordova world and i'm trying to migrate to capacitor.
Cordova provide these options for ios: 
- StatusBarBackgroundColor
- StatusBarStyle
- StatusBarOverlaysWebView
The first two have been implemented in capacitor (StatusBar setStyle and setBackgroundColor) but the last one is not available.

Thank you

**Comment by jcesarmobile** [comment]
Your message is contradictory, what I mean has nothing to do with StatusBarOverlaysWebView

**Comment by knona** [comment]
What is contradictory ? With cordova, the StatusBarOverlaysWebView option do what you described (i.e. web content is shown below or integrated into the status bar)).

**Comment by jcesarmobile** [comment]
It’s a different thing, what I’ve shown are the safe areas, while StatusBarOverlaysWebView is a hack that resizes the webview and we won’t implement that

**Comment by knona** [comment]
Wheter it be safe areas or StatusBarOverlaysWebView option, the goal isn't the same ?
Anyways, the safe areas could be very useful. It would be perfect if you implement it as you said before.

**Comment by knona** [comment]
I now understand what you said about safe areas. Indeed, it would be really usefull if we had an option to change that. On ios, the status bar and home bar are annoying. The only way to simulate safe ares is to play with css.
`https://developer.mozilla.org/en-US/docs/Web/CSS/env`
I'm gonna try this.

**Comment by jcesarmobile** [comment]
Since it looks like the problem was Capacitor disabling content insets, I've added a new configuration option, see https://github.com/ionic-team/capacitor/pull/2392 for more details.

TLDR, add this to `capacitor.config.json` to get the old behaviour. (there are move possible values)

```
"ios": {
      "contentInset": "always"
  }
```

**Comment by knona** [comment]
Thanks !

**Comment by Snake231088** [comment]
I've added "contentInset": "always" in capacitor.config.json and now the page is correctly fit in safe area but when i scroll, the content goes under transparent statusbar.
How i can to solve this issue? Thanks!

**Comment by pradella** [comment]
@Snake231088 set viewport-fit=cover to fix it

**Comment by danbaechtold** [comment]
@jcesarmobile Thanks, "contentInset": "always" is almost perfect, but I have the same issue like @Snake231088: The AppBar (position fixed) now correctly stays below the StatusBar. However when scrolling, the content scrolls over the StatusBar (but below AppBar)!

![image](https://user-images.githubusercontent.com/4057377/131694610-0697402d-feeb-4be8-b3ae-f57c020a2342.png)


And I have set viewport-fit=cover, like @pradella suggested.

Can you further improve what contentInset does, so that this doesn't happen?
Or is there anything I have to do on my end still?

**Comment by dshastry** [comment]
I'm having the same issue and I'm trying to figure it out as well.  Fixed header but content scrolls below it exactly like @danbaechtold  screenshot.

**Comment by chwebdev815** [comment]
I was the exact same issue.
"contentInset": "always"
viewport-fit=cover
But content is overlaying the status bar still.
Could you share the perfect solution?
![image](https://user-images.githubusercontent.com/35192503/141179794-db65569d-9284-4caf-bc05-e6eb8448f20f.png)

**Comment by prasanthmarimi** [comment]
@chwebdev815 please let me know if you found any solution to this issue? Was getting the same in my app.

**Comment by chwebdev815** [comment]
@prasanthmarimi 
Yes, I fixed the issue but used `safearea` library from [capacitor-plugin-safe-area](https://www.npmjs.com/package/capacitor-plugin-safe-area)

For a more detailed, when inset is detected, I manually set insets.insets.top, insets, insets.bottom like this

```
const [appInsets, setAppinsets] = useState({top: 0, bottom:0})

useEffect(() => {
    const setupWithInsets = async function () {
      const insets = await SafeArea.getSafeAreaInsets()
      appInsets.top = insets.insets.top;
      appInsets.bottom = insets.insets.bottom
      setAppInsects({...appInsets})
    } 
    setupWithInsets();
  }, []);```

**Comment by HarelM** [comment]
Seem like setting `always` didn't solve the problem for me.
Has anyone used the @capacitor/status-bar to solve this?
I used the cordova one to fix a similar issue but I want to avoid using cordova plugins as much as possible...
